### ISSUE TITLE
Security information regarding redis

### ISSUE THREAD

**Issue by johnxba** [owncloud/core#39119]
### Current
Documentation for redis does not contain sufficient security/hardening information. It does refer to the [redis documentation](https://redis.io/documentation) but that doesn't provide settings specifically applicable to ownCloud.

### Wanted
Please create elaborate documentation for redis, especially regarding securing the setup and what it needs and doesn't need.

### Server configuration
operating system: Debian 11.0
redis version: 6.0.15
ownCloud version: 10.8.0.4
web server: Apache 2.4.48

Details:
Redis has a page here: [https://doc.owncloud.com/server/10.8/admin_manual/configuration/server/caching_configuration.html#redis-configuration](https://doc.owncloud.com/server/10.8/admin_manual/configuration/server/caching_configuration.html#redis-configuration) for use with owncloud. There is some tuning information [here](https://doc.owncloud.com/server/admin_manual/configuration/server/oc_server_tuning.html#redis-tuning).

Except for password protection and communication channels, it doesn't help to further secure and harden Redis.

Redis can name commands such as FLUSHDB, FLUSHALL, CONFIG, DEBUG (`rename-command`). 
Documentation should have a list of commands that can be renamed/made unavailable. These questions should be answered:
- Which commands are not used during normal day to day operation of ownCloud? 
- Which commands are needed to conduct an ownCloud update? 
- Any special cases?

/etc/redis/redis.conf (v6) contains:
```
# Since Redis 2.6 by default replicas are read-only.
#
# Note: read only replicas are not designed to be exposed to untrusted clients
# on the internet. It's just a protection layer against misuse of the instance.
# Still a read only replica exports by default all the administrative commands
# such as CONFIG, DEBUG, and so forth. To a limited extent you can improve
# security of read only replicas using 'rename-command' to shadow all the
# administrative / dangerous commands.
replica-read-only yes

#disable commands
rename-command CONFIG ""
rename-command FLUSHDB ""
rename-command FLUSHALL ""
rename-command DEBUG ""
```

- Which restrictive command rename settings can be used with ownCloud? 
- Same question for any other security related settings.

Redis has deprecated command renaming in favor of using ACL's:
```
# Command renaming (DEPRECATED).
#
# ------------------------------------------------------------------------
# WARNING: avoid using this option if possible. Instead use ACLs to remove
# commands from the default user, and put them only in some admin user you
# create for administrative purposes.
# ------------------------------------------------------------------------
#
# It is possible to change the name of dangerous commands in a shared
# environment. For instance the CONFIG command may be renamed into something
# hard to guess so that it will still be available for internal-use tools
# but not available for general clients.
#
# Example:
#
# rename-command CONFIG b840fc02d524045429941cc15f59e41cb7be6c52
#
# It is also possible to completely kill a command by renaming it into
# an empty string:
#
#rename-command CONFIG ""
#
# Please note that changing the name of commands that are logged into the
# AOF file or transmitted to replicas may cause problems.
```

Documentation should provide secure ACL's applicable to ownCloud. These questions should be answered:
- What ACL can be set during normal day to day operation of ownCloud? 
- What ACL is needed to conduct an ownCloud update? 
- Any special cases?

Perhaps different users can be prepared for different functions.

redis can be reloaded by `systemctl reload redis`.

Thank you very much for owncloud and reading this.

**Comment by JammingBen** [comment]
@mmattel fyi

**Comment by mmattel** [comment]
Hello @johnxba many thanks for opening this issue.

Please differenciate the following:
1) providing conncetion details for the use of Redis with ownCloud, and
2) administrate Redis

Item 1 is addressed at 

https://doc.owncloud.com/server/latest/admin_manual/configuration/server/caching_configuration.html#redis-configuration
and at https://doc.owncloud.com/server/1latest/admin_manual/configuration/server/config_sample_php_parameters.html#define-redis-connection-details

Item 2 cant be provided by the ownCloud documentation. If we would do this kind of documentation for Redis you request, we would need to do this for Apache and others, which is not the aim of our docs. You request is a classical support for administrative tasks. You have to either train yourself on the topic or use our consulting services, we are happy to help.

**Comment by johnxba** [comment]
@mmattel, I did not ask about connection details, that must be a misunderstanding.

The first link you provided is already in my message. The second link is a 404, but I found the intended link. Both do not contain the asked for security settings (commands needed and ACL's). 

The ownCloud project is the obvious party that should provide these settings. It depends specifically on ownCloud's use of Redis. Nobody else but people contributing to the project know what's used and what will be needed.

Apache specific security settings are self contained and do not depend on ownCloud. The requested Redis security settings do depend on what ownCloud needs. What ownCloud does with or needs from redis is not in the current documentation.

What commands does ownCloud use? What ACL (permissions) does it need for specific Redis work?

**Comment by mmattel** [comment]
@johnxba as you have seen, I updated the documentation. ownCloud currently does not support Redis ACL´s. This is a feature introduced with Redis version 6. At the moment, Ubuntu 20.04 only delivers Redis and the php-redis extension in version 5. Thanks for catching this. Just to let you know, none of our customers requested this feature and all were happy using the password option, so we did not add any effort into implementing it.
